docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,255 @@
+to help you understand the nuances of Gno and how to use it effectively.
+
+Before we dive in, it's important to note that Gno shares several similarities
+with Go. Therefore, if you haven't already, we highly recommend reading
+["Effective Go"](https://go.dev/doc/effective_go) as a primer.
+
+## Counter-Intuitive Good Practices
+
+This section highlights some Gno good practices that might seem
+counter-intuitive, especially if you're coming from a Go background.
+
+### Embrace Global Variables in Gno
+
+In Gno, using global variables isn't just acceptable - it's encouraged. This is
+because global variables in Gno provide a way to have persisted states
+automatically.
+
+In Go, you would typically write your logic and maintain some state in memory.
+However, to persist the state and ensure it survives a restart, you would need
+to use a store (like a plain file, custom file structure, a database, a
+key-value store, an API, etc.).
+
+In contrast, Gno simplifies this process. When you declare global variables in
+Gno, the GnoVM automatically persists and restores them as needed between each
+run.
+
+However, be mindful not to export your global variables. Doing so would make
+them accessible for everyone to read and write.
+
+Here's an ideal pattern to follow:
+
+```go
+var counter int
+
+func GetCounter() int {
+    return counter
+}
+
+func IncCounter() {
+    counter++
+}
+```
+
+### Embrace Panic in Gno
+
+In Gno, it's important to know when to return an `error` and when to use `panic()`.
+Each does something different to your code and data.
+
+When you return an error in Gno, it's like giving back any other piece of data.
+It tells you something went wrong, but it doesn't stop your code or undo any
+changes you made.
+
+But, when you use panic in Gno, it stops your code right away, says it failed,
+and doesn't save any changes you made. This is safer when you want to stop
+everything and not save wrong changes.
+
+In general, it's good to use `panic()` in realms. In reusable packages, you can
+use either panic or errors, depending on what you need.
+
+```go
+import "std"
+
+func Foobar() {
+    caller := std.GetOrigCaller()
+    if caller != "g1234567890123456789012345678912345678" {
+        panic("permission denied")
+    }
+    // ...
+}
+```
+
+- TODO: suggest MustXXX and AssertXXX flows in p/.
+
+### Understand the importance of `init()`
+
+In Gno, the `init()` function isn't just a function, it's a cornerstone. It's
+automatically triggered when a new realm is added onchain, making it a one-time
+setup tool for the lifetime of a realm.
+
+Unlike Go, where `init()` is used for tasks like setting up database
+connections, configuring logging, or initializing global variables every time
+you start a program, in Gno, `init()` is executed once per realm's lifetime.
+
+In Gno, `init()` primarily serves two purposes:
+1. It registers your new realm on a new realm. This is typically done using the
```

Comment:
> It registers your new realm on a new realm.

This is confusing wordage. I am not sure what you are trying to say though perhaps there is a way to restate this- like creating an instance of another realm within this realm in order to call its corresponding methods?

Reply:
c63a170c